python/mxnet/kvstore.py

```diff
@@ -349,7 +349,17 @@ def row_sparse_pull(self, key, out=None, priority=0, row_ids=None):
         """
         assert(out is not None)
         assert(row_ids is not None)
-        ckeys, cvals, use_str_keys = _ctype_key_value(key, out)
+        if isinstance(row_ids, NDArray):
+            row_ids = [row_ids]
+        assert(isinstance(row_ids, list)), \
+            "row_ids should be NDArray or list of NDArray"
+        out_val = out
```

Comment:
I'd prefer `first_out` to `out_val`. I also recommend document the optimization **upfront** instead of and the end of the fucntion: 
"When there is only one row_id, we can invoke KVStoreRowSparsePull just once and broadcast the result to all the rest of outputs"

Reply:
comments are added into doc string